docs/api-communication-spec.md

```diff
@@ -0,0 +1,686 @@
+        subgraph cf-app2
+            pcap-api -->|H/2, gRPC| c2e[Envoy]
+            c2e -->|gRPC, H2C| c2[pcap-agent]
+        end 
+    end
+    subgraph diego-cell2
+        subgraph cf-app3
+            pcap-api -->|H/2, gRPC| c3e[Envoy]
+            c3e -->|H2C, gRPC| c3[pcap-agent]
+        end 
+    end
+    subgraph cloud-controller
+        pcap-api -->|HTTP/1.1, JSON| cc[Cloud Controller]
+    end
+```
+Figure 1: The Cloud Foundry Application Capturing case
+
+## BOSH Case Overview
+
+Capturing data from BOSH deployments requires the pcap-agent on the individual BOSH VMs. The pcap-agent runs on a well-known port and takes care of mTLS verification.
+
+The BOSH director is used to verify and find the requested target VMs' IP addresses. The BOSH UAA is used to verify the token provided by the client.
+
+```mermaid
+graph LR
+    pcap-cli -->|H/2, gRPC| pcap-api
+
+    subgraph pcap deployment
+        pcap-api
+    end
+        
+    subgraph bosh-deployment-vm1
+        pcap-api -->|H/2, gRPC| a1[pcap-agent]
+    end
+    subgraph bosh-deployment-vm2
+        pcap-api -->|H/2, gRPC| a2[pcap-agent]
+    end
+
+    subgraph bosh-director-vm
+        pcap-api -->|HTTP/1.1, JSON| bosh[BOSH Director]
+        pcap-api -->|HTTP/1.1, JSON| bosh-uaa[BOSH UAA]
+    end
+```
+
+Figure 2: The BOSH Capturing case
+
+## Interactions
+
+### pcap-cli -> pcap-api
+  - request status (version information, service alive)
+  - capture request for:
+    - CF Capture Request
+    - BOSH Capture Request
+  - Send stop request
+### pcap-api
+  - Status response
+  - CF Capture Request
+    - pcap-api -> Cloud Controller
+      - Validate request, token
+      - find endpoints (pcap-agent)
+  - BOSH Capture Request
+    - pcap-api -> BOSH Director
+      - HTTP: Retrieve /info (also works without Auth)
+      - HTTP: Retrieve deployment instances
+    - pcap-api -> BOSH Director UAA
+      - HTTP: Retrieve JWT token key
+  - pcap-api -> pcap-agent
+    - Status
+    - Send Capture Request
+    - Send stop request
+### pcap-agent -> pcap-api
+  - Send status
+  - Send Capture Response
+    - Message
+      - Stop
+      - Congestion
+      - Error
+        - Limit reached (e.g. number of concurrent capture requests exceeded, )
+        - Invalid Request (e.g. malformed filter, device not found, etc.)
+        - Runtime (any unexpected error occurring on the agent during capture. Contains as much information as possible and forward before terminating)
+    - captured pcap data packet(s) - consider batching multiple packets in one Capture Response
+### pcap-api -> pcap-cli
+
+- Status
+- captured pcap data packet
+- Message
+    - Stop confirmation
+    - Capture status
+        - Agent is gone
+        - All agents stopped
+        - Congestion
+    - Errors
+        - Limit reached (e.g. number of concurrent capture requests exceeded, )
+        - Unauthorized
+        - Invalid Request (e.g. malformed filter, Target not found, device not found, etc.)
```

Comment:
AFAIR you told me at one point not to use `e.g.` and `etc.` because they express the same thing: a non-exhaustive list :P